services/api-server/tests/unit/conftest.py

```diff
@@ -0,0 +1,236 @@
+    assert folder.exists()
+    assert any(folder.glob("src/simcore_service_api_server"))
+    return folder
+
+
+@pytest.fixture(scope="session")
+def package_dir():
+    dirpath = Path(simcore_service_api_server.__file__).resolve().parent
+    assert dirpath.exists()
+    return dirpath
+
+
+@pytest.fixture(scope="session")
+def osparc_simcore_root_dir(project_slug_dir):
+    root_dir = project_slug_dir.parent.parent
+    assert (
+        root_dir and root_dir.exists()
+    ), "Did you renamed or moved the integration folder under api-server??"
+    assert any(root_dir.glob("services/api-server")), (
+        "%s not look like rootdir" % root_dir
+    )
+    return root_dir
+
+
+@pytest.fixture(scope="session")
+def tests_dir() -> Path:
+    tdir = (current_dir / "..").resolve()
+    assert tdir.exists()
+    assert tdir.name == "tests"
+    return tdir
+
+
+@pytest.fixture(scope="session")
+def tests_utils_dir(tests_dir: Path) -> Path:
+    utils_dir = (tests_dir / "utils").resolve()
+    assert utils_dir.exists()
+    return utils_dir
+
+
+## POSTGRES & APP ---
+
+
+@pytest.fixture(scope="session")
+def docker_compose_file(environment, tests_utils_dir, tmpdir_factory) -> Path:
+    # Overrides fixture in https://github.com/avast/pytest-docker
+
+    src_path = tests_utils_dir / "docker-compose.yml"
+    assert src_path.exists
+
+    dst_path = Path(str(tmpdir_factory.mktemp("config").join("docker-compose.yml")))
+
+    shutil.copy(src_path, dst_path.parent)
+    print(list(dst_path.parent.glob("*")))
+
+    # configs
+    subprocess.run(
+        f"docker-compose --file {src_path} config > {dst_path.name}",
+        shell=True,
+        check=True,
+        cwd=dst_path.parent,
+        env=environment,
+    )
+
+    return dst_path
+
+
+@pytest.fixture(scope="session")
+def postgres_service(docker_services, docker_ip, docker_compose_file: Path) -> Dict:
+
+    # check docker-compose's environ is resolved properly
+    config = yaml.safe_load(docker_compose_file.read_text())
+    environ = config["services"]["postgres"]["environment"]
+
+    # builds DSN
+    config = dict(
+        user=environ["POSTGRES_USER"],
+        password=environ["POSTGRES_PASSWORD"],
+        host=docker_ip,
+        port=docker_services.port_for("postgres", 5432),
+        database=environ["POSTGRES_DB"],
+    )
+
+    dsn = "postgresql://{user}:{password}@{host}:{port}/{database}".format(**config)
+
+    def _create_checker() -> Callable:
+        def is_postgres_responsive() -> bool:
+            try:
+                engine = sa.create_engine(dsn)
+                conn = engine.connect()
+                conn.close()
+            except sa.exc.OperationalError:
+                return False
+            return True
+
+        return is_postgres_responsive
+
+    # Wait until service is responsive.
+    docker_services.wait_until_responsive(
+        check=_create_checker(), timeout=30.0, pause=0.1,
+    )
+
+    config["dsn"] = dsn
+    return config
+
```

Comment:
use pytest-simcore instead?

Reply:
I did not want to bring in any aiohttp dependency ... but perhaps you are right.